src/bio/bio_xstream.c

```diff
@@ -83,18 +83,27 @@ bio_spdk_env_init(void)
 
 	spdk_env_opts_init(&opts);
 	opts.name = "daos";
+
 	/*
 	 * TODO: Set opts.mem_size to nvme_glb.bd_mem_size
 	 * Currently we can't guarantee clean shutdown (no hugepages leaked).
 	 * Setting mem_size could cause EAL: Not enough memory available error,
 	 * and DPDK will fail to initialize.
 	 */
 
-	if (nvme_glb.bd_shm_id != DAOS_NVME_SHMID_NONE)
+	if (nvme_glb.bd_shm_id == DAOS_NVME_SHMID_NONE)
 		opts.shm_id = nvme_glb.bd_shm_id;
 
-	/* quiet DPDK logging by setting level to ERROR */
-	opts.env_context = "--log-level=lib.eal:4";
+	/*
+	 * TODO: Find a way to set multiple overrides, currently only single
+	 * option can be overridden with opts.env_context.
+	 */
+
+	/*
+	 * Quiet DPDK logging by setting level to ERROR
+	 *  opts.env_context = "--log-level=lib.eal:4";
+	 */
+	opts.env_context = "--no-telemetry";
```

Comment:
opts.env_context = "--log-level=lib.eal:4 --no-telemetry" doesn't work?

Reply:
no, only a single option can be overridden using env_context, details added to the ticket